liquid-rust-typeck/src/bblock_env.rs

```diff
@@ -11,20 +11,21 @@ use liquid_rust_lrir::{
 };
 
 pub struct BBlockEnv {
-    pub ghost_vars: OrderedHashMap<GhostVar, Ty>,
+    pub ghost_vars: BTreeMap<GhostVar, Ty>,
     pub locals: Vec<(Local, GhostVar)>,
 }
 
 impl BBlockEnv {
     pub fn new<'tcx>(
-        local_decls: &IndexMap<Local, LocalDecl<'tcx>>,
+        local_decls: &IndexVec<Local, LocalDecl<'tcx>>,
         mut refiner: Refiner<'_, 'tcx>,
         vars_in_scope: &mut Vec<Var>,
+        ghost_gen: &IndexGen<GhostVar>,
     ) -> Self {
-        let mut ghost_vars = OrderedHashMap::new();
+        let mut ghost_vars = BTreeMap::new();
         let mut locals = vec![];
         for (local, local_decl) in local_decls.iter_enumerated() {
-            let fresh_gv = refiner.var_gen.fresh();
+            let fresh_gv = ghost_gen.fresh();
```

Comment:
I wasn't sure if it was necessary to use the atomic in the index generator, but this convinced me :+1:

Reply:
Me neither! I just did it beforehand because I didn't want to risk requiring a mutable reference somewhere where it wasn't available hahaha.